File: 01-calorie-counting/python/main.py
with open("../input.txt", "r") as f:
    max_calories, current_calories = 0, 0
    for line in f:
        if line != "\n":
            current_calories += int(line)
        else:
            max_calories = max(max_calories, current_calories)
            current_calories = 0
    print(max_calories)

File: 01-calorie-counting/node/main.js
const fs = require("fs");

let maxCalories = 0;
let currentCalories = 0;

try {
  const fileContent = fs.readFileSync("../input.txt", "utf-8");
  fileContent.split(/\r?\n/).forEach((line) => {
    if (line !== "") {
      currentCalories += parseInt(line);
    } else {
      if (maxCalories < currentCalories) {
        maxCalories = currentCalories;
      }
      currentCalories = 0;
    }
  });
  console.log(maxCalories);
} catch (err) {
  console.error(err);
}
